### DISCUSSION TITLE
Add padding for Batteryicon widget

### DISCUSSION THREAD

**Discussion by MagnuSiwy** [qtile/qtile#4725] [Ideas] [answered]
When using rounded corners from qtile-extras it's more visible, but the problem can be easily seen in the default qtile config as well. 

The changed icons don't fit in the widget. I think if they had their own padding it could be avoided. The problem persists when using the default set of icons as well. Doesn't really matter if it's a .png or .svg file.

![image](https://github.com/qtile/qtile/assets/84531183/73cf943b-81c6-4da4-bf5d-17830c4a7a10)

**Comment by elParaguayo** [comment]
In the meantime, you can insert a `Spacer` widget before the icon and give it a small, fixed width.

**Reply by MagnuSiwy** [reply]
I did that, but unfortunately it doesn't get grouped automatically. In my previous version I was using the `Battery` widget and changed the "chars". I will check how to group elements manually though

**Reply by elParaguayo** [reply]
It should get grouped. Did you add the decoration to the Spacer too?

**Reply by MagnuSiwy** [reply]
Yeah, I did and it looked like that even when I changed the lenght at first
![image](https://github.com/qtile/qtile/assets/84531183/861487be-10be-42ad-9f31-1b4d12464131)

I set it to the lenght 10 (that's the default size of the padding in my config) and it worked. Maybe it just had to be longer than the `BatteryIcon` widget is?

**Reply by elParaguayo** [reply]
It would need to be wider than the radius of the rounded corners (which you wouldn't need to worry about if the `BatteryIcon` widget had the ability to adjust its `padding`).

**Reply by MagnuSiwy** [reply]
Ohh, yeah, the corners radius is set to 10 as well. Ok, that should temporarily work well. Thanks!

**Reply by elParaguayo** [reply, marked as answer]
https://github.com/qtile/qtile/pull/4727
